torch/cuda/__init__.py

```diff
@@ -317,41 +318,93 @@ def can_device_access_peer(device: _device_t, peer_device: _device_t) -> bool:
         raise AssertionError("Invalid peer device id")
     return torch._C._cuda_canDeviceAccessPeer(device, peer_device)
 
-
-@contextlib.contextmanager
-def stream(stream):
+class StreamContext(object):
     r"""Context-manager that selects a given stream.
```

Comment:
It seems that this comment has been copied from the old implementation. I think a lot of it still applies but needs to be edited or moved (e.g. to `__init__`) since this is a class and not a function.